### DISCUSSION TITLE
Basic example for grpc/grpc-web?

### DISCUSSION THREAD

**Discussion by louisrli** [stephenh/ts-proto#776] [Q&A]
I played around with generating client code for the @improbable-eng/grpc-web (which is no longer maintained), but I was wondering if there's a similar way to generate a client for the official grpc/grpc-web (I'm trying to figure out the client-side code from the browser)

I'm trying to follow along with the example code for calling from a client here:
https://github.com/grpc/grpc-web/tree/master/net/grpc/gateway/examples/helloworld#write-client-code

I'm currently generating the protos without any flags:
```
protoc --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=client/src proto/*.proto
```

But perhaps I'm missing something? I see a generated class called `SheetsServiceClientImpl`, though it seems to take an `rpc` object and I'm having trouble finding an example of how to do this.

**Comment by stephenh** [comment]
Hi @louisrli ! That's a good question; we don't support that right now, but it'd probably be possible? I guess?

I haven't personally used grpc-web, so I've generally had to rely on others to implement the grpc-web support within ts-proto.

If you'd like to poke around and migrate/upgrade ts-proto's output to work with grpc.js's grpc-web instead of @improbable-eng, that'd be great!
